pkg/clusterresource/ibmcloud.go

```diff
@@ -0,0 +1,98 @@
+package clusterresource
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	installertypes "github.com/openshift/installer/pkg/types"
+	installeribmcloud "github.com/openshift/installer/pkg/types/ibmcloud"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	hivev1ibmcloud "github.com/openshift/hive/apis/hive/v1/ibmcloud"
+	"github.com/openshift/hive/pkg/constants"
+)
+
+var _ CloudBuilder = (*IBMCloudBuilder)(nil)
+
+// IBMCloudBuilder encapsulates cluster artifact generation logic specific to IBM Cloud.
+type IBMCloudBuilder struct {
+	// AccountID is the IBM Cloud Account ID
+	AccountID string `json:"accountID"`
+
+	// CISInstanceCRN is the IBM Cloud Internet Services Instance CRN
+	CISInstanceCRN string `json:"cisInstanceCRN"`
+
+	// APIKey is the IBM Cloud api key.
+	APIKey string
+
+	// Region specifies the IBM Cloud region where the cluster will be
+	// created.
+	Region string `json:"region"`
+
+	// InstanceType specifies the IBM Cloud instance type
+	InstanceType string `json:"instanceType"`
+}
+
+func (p *IBMCloudBuilder) GenerateCredentialsSecret(o *Builder) *corev1.Secret {
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      p.CredsSecretName(o),
+			Namespace: o.Namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		StringData: map[string]string{
+			// This API KEY will be passed to the installer as constants.IBMCloudAPIKeyEnvVar
+			constants.IBMCloudAPIKeySecretKey: p.APIKey,
+			// IBMCloud libraries support reading different auth from environment using GetAuthenticatorFromEnvironment
+			// ref: https://github.com/IBM/cloudant-go-sdk/blob/931e800d66b843b79a23b439fec43a90509c649e/common/cloudant_base.go#L171
+			// This secret key is currently not passed to the installer pod's environment but may be in the future
+			constants.IBMCloudCredentialsEnvSecretKey: fmt.Sprintf("IBMCLOUD_AUTHTYPE=iam\nIBMCLOUD_APIKEY=%s", p.APIKey),
```

Comment:
This can probably be removed until something will actually use this format for IBMCloud (basically wait for ManagedDNS support for IBMCloud to introduce this). But it's harmless as well, so whatever makes the most sense.

Reply:
I'll remove this one for now.